src/main/java/com/redhat/jenkins/plugins/ci/pipeline/CIMessageSubscriberStep.java

```diff
@@ -136,54 +134,43 @@ public StepExecution start(StepContext context) throws Exception {
             if (step.getProviderName() == null) {
                 throw new Exception("providerName not specified!");
             }
 
-            GlobalCIConfiguration config = GlobalCIConfiguration.get();
-            JMSMessagingProvider provider = config.getProvider(step.providerName);
-            if (provider == null) {
-                throw new RuntimeException("Failed to locate JMSMessagingProvider with name "
-                        + step.providerName + ". You must update the job configuration.");
-            }
-
-            watcher = provider.createWatcher();
-            int timeout = CIMessageSubscriberBuilder.DEFAULT_TIMEOUT_IN_MINUTES;
-            if (step.getTimeout() != null && step.getTimeout() > 0) {
-                timeout = step.getTimeout();
-            }
-            watcher.setTimeout(timeout);
-            watcher.setOverrides(step.overrides);
-            watcher.setSelector(step.selector);
-            watcher.setChecks(step.checks);
-            watcher.setProvider(provider);
-            watcher.setEnvironment(getContext().get(Run.class).getEnvironment(getContext().get(TaskListener.class)));
-            watcher.setTaskListener(getContext().get(TaskListener.class));
-
             task = Timer.get().submit(new Runnable() {
                 @Override
                 public void run() {
-                    String msg = watcher.watch();
-                    if (msg == null) {
-                        getContext().onFailure(new AbortException("Timeout waiting for message!"));
-                    } else {
-                        getContext().onSuccess(msg);
+                    try {
+                        CIMessageSubscriberBuilder subscriber = new CIMessageSubscriberBuilder(
+                                step.getProviderName(),
+                                step.getOverrides(),
+                                step.getSelector(),
+                                step.getChecks(),
+                                step.getTimeout()
+                                );
+                        StepContext c = getContext();
+                        String result = subscriber.waitforCIMessage(c.get(Run.class), c.get(Launcher.class), c.get(TaskListener.class));
+                        if (result != null) {
+                            getContext().onSuccess(result);
+                        } else {
+                            getContext().onFailure(new AbortException("Timeout waiting for message!"));
+                        }
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        getContext().onFailure(e);
                     }
-                }
+}
             });
             return false;
         }
 
         @Override
         public void stop(@Nonnull Throwable cause) throws Exception {
             if (task != null) {
-                getContext().get(TaskListener.class).getLogger().println("in stop of watcher");
```

Comment:
why is this being removed?

Reply:
To me it appears to be a debug message that got left in accidentally.  I can put it back.